test/test_rosbag/bag_migration_tests/CMakeLists.txt

```diff
@@ -5,22 +5,82 @@
 # generate_messages() it might make it work, or we can migrate the bag
 # files themselves.
 
-add_message_files(
-  DIRECTORY msg_current
-  FILES
-  Constants.msg
-  Converged.msg
-  MigratedAddSub.msg
-  MigratedExplicit.msg
-  MigratedImplicit.msg
-  MigratedMixed.msg
-  PartiallyMigrated.msg
-  Renamed4.msg
-  Simple.msg
-  SimpleMigrated.msg
-  SubUnmigrated.msg
-  Unmigrated.msg
-)
+set(MSG_DIRECTORY "msg_current" CACHE STRING "Messages directory")
+message(STATUS "Building messages in: ${MSG_DIRECTORY}")
+
+if(${MSG_DIRECTORY} STREQUAL "msg_gen1")
+  add_message_files(
+    DIRECTORY ${MSG_DIRECTORY}
+    FILES
+    Constants.msg
+    Converged.msg
+    Convergent.msg
+    MigratedAddSub.msg
+    MigratedExplicit.msg
+    MigratedImplicit.msg
+    MigratedMixed.msg
+    PartiallyMigrated.msg
+    Renamed1.msg
+    Simple.msg
+    SimpleMigrated.msg
+    SubUnmigrated.msg
+    Unmigrated.msg
+  )
```

Comment:
`add_message_files` can be called more than once. Would it make sense to have a single block for all the common ones, and then only do conditionally the ones that are different (eg `Renamed1`/`Renamed2`)?

Note also that CMake lets you do fun stuff like this:

```
set(msglist_msg_gen1 Constants.msg Converged.msg ...)
add_message_files(
  DIRECTORY ${MSG_DIRECTORY}
  FILES ${msglist_${MSG_DIRECTORY}}
)
```